src/Raven.Server/Documents/Queries/AST/JavascriptCodeQueryVisitor.cs

```diff
@@ -95,9 +96,92 @@ public override void VisitCompoundWhereExpression(BinaryExpression @where)
 
         public override void VisitMethod(MethodExpression expr)
         {
+            // The regex usage here is just temporary, to get things working. Ideally, we want to implement the string functions in Jint
+            if (expr.Name.Value.Equals("startswith", StringComparison.OrdinalIgnoreCase) && expr.Arguments.Count == 2)
+            {
+                _sb.Append("new Regexp('^");
```

Comment:
Javascript _has_ startsWith method

Reply:
https://github.com/sebastienros/jint/issues/429

Reply:
Use polyfill:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

Reply:
However, regardless, we need to implement it directly.